### PR TITLE
adding Sofy integration demo project

### PR DESCRIPTION
**Pull request by icarusdust**

### CODE REVIEW

File: integrations/sofy_integration_demo_project/codemagic.yaml [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+      - name: Flutter analyze
+        script: |
+          cd . && flutter analyze
+      - name: Flutter unit tests
+        script: |
+          cd . && flutter test
+        ignore_failure: true          
+      - name: Build APK with Flutter  
+        script: |
+          cd . && flutter build apk --release --build-name=1.0.0 --build-number=$(($(google-play get-latest-build-number --package-name "$PACKAGE_NAME" --tracks="$GOOGLE_PLAY_TRACK") + 1))
```
Comment (nevercode-kevin) [inline integrations/sofy_integration_demo_project/codemagic.yaml:48]:
```suggestion
      - name: Get Flutter packages
        script: |
          flutter packages pub get
      - name: Flutter analyze
        script: |
          flutter analyze
      - name: Flutter unit tests
        script: |
          flutter test
        ignore_failure: true          
      - name: Build APK with Flutter  
        script: |
          flutter build apk --release --build-name=1.0.0 --build-number=$(($(google-play get-latest-build-number --package-name "$PACKAGE_NAME" --tracks="$GOOGLE_PLAY_TRACK") + 1))
```

File: integrations/sofy_integration_demo_project/codemagic.yaml [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        # Add the group environment variables in Codemagic UI (either in Application/Team variables) - https://docs.codemagic.io/variables/environment-variable-groups/
+        - keystore_credentials # <-- (Includes CM_KEYSTORE, CM_KEYSTORE_PASSWORD, CM_KEY_PASSWORD, CM_KEY_ALIAS)
+          # CM_KEYSTORE <-- Put your keystore file here
+          # CM_KEYSTORE_PASSWORD <-- Put your keystore password here
+          # CM_KEY_PASSWORD <-- Put your keystore alias password here
+          # CM_KEY_ALIAS <-- Put your keystore alias username here
+        - google_play # <-- (Includes GCLOUD_SERVICE_ACCOUNT_CREDENTIALS - Put your google-services.json here)
+        - other
+      vars:
+        FCI_KEYSTORE_PATH: /tmp/keystore.keystore
```
Comment (nevercode-kevin) [inline integrations/sofy_integration_demo_project/codemagic.yaml:16]:
```suggestion
        CM_KEYSTORE_PATH: /tmp/keystore.keystore
```

File: integrations/sofy_integration_demo_project/codemagic.yaml [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+      cocoapods: default
+    scripts:
+      - name: Set up key properties
+        script: |
+          echo $CM_KEYSTORE | base64 --decode > /tmp/keystore.keystore
+          # The following script creates key.properties file and stores the credentials in it. As we configure code signing in app/build.gradle file, the following part is unnecessary unless code signing is configured differently as explained in the documentation: https://docs.codemagic.io/flutter-code-signing/android-code-signing/
+          # cat >> "$CM_BUILD_DIR/android/key.properties" <<EOF
+          # storePassword=$FCI_KEYSTORE_PASSWORD
+          # keyPassword=$FCI_KEY_PASSWORD
+          # keyAlias=$FCI_KEY_ALIAS
```
Comment (nevercode-kevin) [inline integrations/sofy_integration_demo_project/codemagic.yaml:30]:
```suggestion
          # storePassword=$CM_KEYSTORE_PASSWORD
          # keyPassword=$CM_KEY_PASSWORD
          # keyAlias=$CM_KEY_ALIAS
```

File: integrations/sofy_integration_demo_project/codemagic.yaml [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+      - name: Get Flutter packages
+        script: |
+          cd . && flutter packages pub get
+      - name: Flutter analyze
+        script: |
+          cd . && flutter analyze
+      - name: Flutter unit tests
+        script: |
+          cd . && flutter test
+        ignore_failure: true
```
Comment (nevercode-kevin) [inline integrations/sofy_integration_demo_project/codemagic.yaml:138]:
```suggestion
      - name: Get Flutter packages
        script: |
          flutter packages pub get
      - name: Flutter analyze
        script: |
          flutter analyze
      - name: Flutter unit tests
        script: |
          flutter test
        ignore_failure: true          
```

File: integrations/sofy_integration_demo_project/codemagic.yaml [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+      email:
+        recipients:
+          - user_1@example.com
+          - user_2@example.com
+        notify:
+          success: true     # To not receive a notification when a build succeeds
+          failure: false    # To not receive a notification when a build fails
+      slack: 
+        # See the following link about how to connect your Slack account - https://docs.codemagic.io/publishing-yaml/distribution/#slack
+        channel: '#builds'
```
Comment (nevercode-kevin) [inline integrations/sofy_integration_demo_project/codemagic.yaml:86]:
```suggestion
        channel: "#builds"
```

**Review by nevercode-kevin — CHANGES_REQUESTED** [review]

**Comment by icarusdust** [comment]
Done!

File: integrations/sofy_integration_demo_project/codemagic.yaml [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+        CM_KEYSTORE_PATH: /tmp/keystore.keystore
+        PACKAGE_NAME: "io.codemagic.flutteryaml" # <-- Put your package name here
+        GOOGLE_PLAY_TRACK: "alpha" # <-- This must be "alpha" or above.  
+      flutter: stable
+      xcode: latest
+      cocoapods: default
+    scripts:
+      - name: Set up key properties
+        script: |
+          echo $CM_KEYSTORE | base64 --decode > /tmp/keystore.keystore
```
Comment (nevercode-kevin) [inline integrations/sofy_integration_demo_project/codemagic.yaml:25]:
```suggestion
          echo $CM_KEYSTORE | base64 --decode > $CM_KEYSTORE_PATH
```

File: integrations/sofy_integration_demo_project/codemagic.yaml [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+          # The following script creates key.properties file and stores the credentials in it. As we configure code signing in app/build.gradle file, the following part is unnecessary unless code signing is configured differently as explained in the documentation: https://docs.codemagic.io/flutter-code-signing/android-code-signing/
+          # cat >> "$CM_BUILD_DIR/android/key.properties" <<EOF
+          # storePassword=$CM_KEYSTORE_PASSWORD
+          # keyPassword=$CM_KEY_PASSWORD
+          # keyAlias=$CM_KEY_ALIAS
+          # storeFile=/tmp/keystore.keystore
```
Comment (nevercode-kevin) [inline integrations/sofy_integration_demo_project/codemagic.yaml:31]:
```suggestion
          # storeFile=$CM_KEYSTORE_PATH
```

File: integrations/sofy_integration_demo_project/codemagic.yaml [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+            --form "applicationFile=@"$APK_PATH""
+      - name: Schedule an automation test with Sofy
+        script: |
+           curl --location --request POST 'https://api.sofy.ai/api/CICD/ScheduleAutomatedTestRun' 
+            --header 'Content-Type: application/json'    
+            --data-raw '{
+            "APIKey":"$SOFY_API_KEY",
+            "ScheduledID":9999
+            }'
+      - name: Schedule an automation test with Sofy
```
Comment (nevercode-kevin) [inline integrations/sofy_integration_demo_project/codemagic.yaml:161]: The name of this script step is the same as the previous one. I see that it uses a different endpoint, so rename the s cript to show that something different is being done.

**Review by nevercode-kevin — CHANGES_REQUESTED** [review]

**Comment by icarusdust** [comment]
Changes made

**Review by nevercode-kevin — APPROVED** [review]
LGTM

**Pull request merged**